tests/template_tests/syntax_tests/test_querystring.py

```diff
@@ -65,6 +74,65 @@ def test_querystring_remove_nonexistent(self):
         output = template.render(context)
         self.assertEqual(output, "?x=y&amp;a=1")
 
+    @setup({"querystring_same_arg": "{% querystring a=1 a=2 %}"})
+    def test_querystring_same_arg(self):
+        with self.assertRaises(TemplateSyntaxError):
+            self.engine.render_to_string("querystring_same_arg", {})
+
+    @setup({"querystring_variable": "{% querystring a=a %}"})
+    def test_querystring_variable(self):
+        request = self.request_factory.get("/")
+        template = self.engine.get_template("querystring_variable")
+        context = RequestContext(request, {"a": 1})
+        output = template.render(context)
+        self.assertEqual(output, "?a=1")
+
+    @setup({"querystring_dict": "{% querystring my_dict %}"})
+    def test_querystring_dict(self):
+        context = {"my_dict": {"a": 1}}
+        output = self.engine.render_to_string("querystring_dict", context)
+        self.assertEqual(output, "?a=1")
+
+    @setup({"querystring_dict_list": "{% querystring my_dict %}"})
+    def test_querystring_dict_list_values(self):
+        context = {"my_dict": {"a": [1, 2]}}
+        output = self.engine.render_to_string("querystring_dict_list", context)
+        self.assertEqual(output, "?a=1&amp;a=2")
+
+    @setup({"querystring_non_string_dict_keys": "{% querystring my_dict %}"})
+    def test_querystring_non_string_dict_keys(self):
+        context = {"my_dict": {0: 1}}
+        msg = "querystring received non-string dict key: 0"
+        with self.assertRaisesMessage(TemplateSyntaxError, msg):
+            self.engine.render_to_string("querystring_non_string_dict_keys", context)
+
+    @setup({"querystring_non_dict_args": "{% querystring somevar %}"})
+    def test_querystring_non_dict_args(self):
+        context = {"somevar": 0}
+        msg = "'int' object has no attribute 'items'"
+        with self.assertRaisesMessage(AttributeError, msg):
+            self.engine.render_to_string("querystring_non_dict_args", context)
+
+    @setup({"querystring_multiple_args": "{% querystring my_query_dict x=0 %}"})
+    def test_querystring_multiple_args(self):
+        context = {"my_query_dict": QueryDict("a=1&b=2")}
+        output = self.engine.render_to_string("querystring_multiple_args", context)
+        self.assertEqual(output, "?a=1&amp;b=2&amp;x=0")
```

Comment:
I think this test is already covered by `test_querystring_with_explicit_query_dict_and_no_request` and can be removed

Reply:
Oh right